mesonbuild/utils/core.py

```diff
@@ -103,18 +104,33 @@ def merge(self, other: EnvironmentVariables) -> None:
         for method, name, values, separator in other.envvars:
             self.varnames.add(name)
             self.envvars.append((method, name, values, separator))
+            if name in self.unset_vars:
+                self.unset_vars.remove(name)
+        self.unset_vars.update(other.unset_vars)
 
     def set(self, name: str, values: T.List[str], separator: str = os.pathsep) -> None:
         self.varnames.add(name)
         self.envvars.append((self._set, name, values, separator))
+        if name in self.unset_vars:
+            self.unset_vars.remove(name)
+
+    def unset(self, name: str) -> None:
+        if name in self.varnames:
+            self.varnames.remove(name)
+            self.envvars = [(m, n, v, s) for (m, n, v, s) in self.envvars if n != name]
```

Comment:
Why keep building tuples? Wouldn't it make more sense to use `[x for x in self.envvars if x[1] != name]`?

Reply:
(This assumes we want to allow you to do both in one meson.build.)